src/inspection/src/inspection/inspection.cpp

```diff
@@ -0,0 +1,56 @@
+#include "inspection/inspection.hpp"
+
+InspectionMission::InspectionMission() : Node("inspection") {
+  // creates publisher that should yield torque/acceleration/...
+  control_command_publisher =
+  this->create_publisher<fs_msgs::msg::ControlCommand>("control_topic", 10);
+
+  // creates publisher for the flag
+  flag_publisher = this->create_publisher<std_msgs::msg::String>("inspection_state", 10);
+
+  // get velocity data from state estimation
+  rpm_subscription = this->create_subscription<fs_msgs::msg::WheelStates>(
+      "velocity_estimation", 10,
+      std::bind(&InspectionMission::publish_torque_steer, this, std::placeholders::_1));
+}
+
+void InspectionMission::publish_torque_steer(fs_msgs::msg::WheelStates current_rpm) {
+  auto current_time = std::chrono::system_clock::now();
+  auto elapsed_time =  (current_time - initial_time).count();
+  if (elapsed_time < 26000000000) {
```

Comment:
There is a more elegant way to do this:
```suggestion
  if (elapsed_time >= 26000000000 || inspection_over) {
    auto flag = std_msgs::msg::String();
    flag.data = "AS Finished";
    flag_publisher->publish(flag);
    inspection_over = true; // flag to change back to false when go signal is received
  }
```